aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -533,12 +533,22 @@ inline bool check_gpu_sm86_head_dim_128(sdp_params params, bool debug) {
   return true;
 }
 
-inline bool check_requires_grad_and_head_dim_128_and_sm86(sdp_params params, bool debug){
-  // Flash Attention will raise an error in the backward pass if the head_dim size is 128
-  // And the device is not sm80, the other head_dim check catches everything but sm86
-  if (!check_requires_grad(params, false) && !check_gpu_sm86_head_dim_128(params, false)){
-    if (debug){
-      TORCH_WARN("Flash attention currently doesn't support training with head_dim == 128 on sm86.");
+inline bool check_requires_grad_and_head_dim_128_and_sm86_or_greater(
+    sdp_params params,
+    bool debug) {
+  // Flash Attention will raise an error in the backward pass if the head_dim
+  // size is 128 And the device is not sm80, the other head_dim check catches
+  // everything but sm86
+  auto dprops = at::cuda::getCurrentDeviceProperties();
+  bool is_sm86_or_newer = (dprops->major == 8) && (dprops->minor >= 6);
+  bool is_sm90_or_newer = (dprops->major == 9) && (dprops->minor >= 0);
```

Comment:
Why do you need separate `sm90_or_newer`?
```suggestion
  bool is_sm86_or_newer = ((dprops->major == 8) && (dprops->minor >= 6)) || (dprops->major > 8);
```

Reply:
sm_90 should work according to checks in backward (but we can't test)